support/util/imagemetadata.go

```diff
@@ -42,52 +45,26 @@ type RegistryClientImageMetadataProvider struct {
 	OpenShiftImageRegistryOverrides map[string][]string
 }
 
-// ImageMetadata returns metadata for a given image using the given pull secret
-// to authenticate. This lookup uses a cache based on the image digest. If the
-// reference of the image contains a digest (which is the mainline case for images in a release payload),
-// the digest is parsed from the image reference and then used to lookup image metadata in the
-// cache. When the image reference does not contain a digest, a lookup is made to the registry to
-// fetch the digest of the image that the tag refers to. This is because the actual image that the
-// tag is referring to could have changed. Once a digest is obtained, the cache is checked so that
-// no further fetching occurs. Only if both cache lookups fail, the image metadata is fetched and
-// stored in the cache.
+// ImageMetadata returns metadata for a given image using the given pull secret to authenticate.
+// The ICSPs/IDMSs are checked first for overrides and then the cache is checked using the image
+// pull spec. If not found in the cache, the manifest is looked up and added to the cache.
 func (r *RegistryClientImageMetadataProvider) ImageMetadata(ctx context.Context, imageRef string, pullSecret []byte) (*dockerv1client.DockerImageConfig, error) {
 
-	var (
-		repo           distribution.Repository
-		ref            *reference.DockerImageReference
-		parsedImageRef reference.DockerImageReference
-		err            error
-	)
-
-	parsedImageRef, err = reference.Parse(imageRef)
+	parsedImageRef, err := reference.Parse(imageRef)
 	if err != nil {
 		return nil, fmt.Errorf("failed to parse image reference %q: %w", imageRef, err)
 	}
 
-	// There are no ICSPs/IDMSs to process.
-	// That means the image reference should be pulled from the external registry
-	if len(r.OpenShiftImageRegistryOverrides) == 0 {
```

Comment:
Does it makes sense to look for an override if there is not any OpenShiftImageRegistryOverrides?

Reply:
The name of the method `seekOverride` indicates that we are looking for an override and may not find one. Looking at the function, it does return the same image reference that was passed into the function if the length of`OpenShiftImageRegistryOverrides` is zero. Therefore, I think checking for the length of the override list at this point only complicates the code.

Reply:
Humm, but it's a return early measure. Did you test it in a real env?

Reply:
Yes, I tested it with in our performance environment with 500+ clusters with `OpenShiftImageRegistryOverrides == 0`.

Reply:
Testing was done in conjunction with https://issues.redhat.com/browse/OCPBUGS-53259, see comment there.